### PR TITLE
Add Xrayreceiver to the integration test

### PR DESCRIPTION
**Pull request by JohnWu20**
Add Xray receiver to the the test frame work, passing the local ECS test.

### CODE REVIEW

File: terraform/ecs/main.tf [hunk 1]
```diff
@@ -170,7 +170,7 @@ resource "aws_ecs_service" "aoc" {
 
   provisioner "local-exec" {
     working_dir = "../../"
-    command = "${module.common.validator_path} --args='-c ${var.validation_config} -t ${module.common.testing_id} --region ${var.region} --metric-namespace ${module.common.otel_service_namespace}/${module.common.otel_service_name} --endpoint http://${aws_lb.aoc_lb[0].dns_name}:${module.common.sample_app_lb_port}'"
+    command = "${module.common.validator_path} --args='-c ${var.validation_config} -t ${module.common.testing_id} --region ${var.region} --metric-namespace ${module.common.otel_service_namespace}/${module.common.otel_service_name} --endpoint http://${aws_lb.aoc_lb[0].dns_name}'"
```
Comment (wyTrivail) [inline terraform/ecs/main.tf:173]: this won't work for other tests, i think we still need to use 8080

File: terraform/template/ec2-docker-compose-config/default_ec2_docker_compose.yml.tpl [hunk 2]
```diff
@@ -9,6 +9,8 @@ services:
       OTEL_RESOURCE_ATTRIBUTES: ${otel_resource_attributes}
       INSTANCE_ID: ${testing_id}
       OTEL_EXPORTER_OTLP_ENDPOINT: ${otel_endpoint}
+      AWS_REGION: ${region}
```
Comment (wyTrivail) [inline terraform/template/ec2-docker-compose-config/default_ec2_docker_compose.yml.tpl:12]: no need to add this i reckon, depends on how you init your aws sdk in your sample app

File: terraform/template/ecstaskdef/xrayreceiver_ecs_taskdef.tpl [hunk 3]
```diff
@@ -20,6 +20,10 @@
         {
             "name": "AWS_XRAY_DAEMON_ADDRESS",
             "value": "127.0.0.1:2000"
+        },
+        {
+            "name": "AWS_REGION",
```
Comment (wyTrivail) [inline terraform/template/ecstaskdef/xrayreceiver_ecs_taskdef.tpl:25]: same as above

File: terraform/template/eks-pod-config/xray-eks-config.tpl [hunk 4]
```diff
@@ -0,0 +1,4 @@
+sample_app:
```
Comment (wyTrivail) [inline terraform/template/eks-pod-config/xray-eks-config.tpl:1]: why not use the existing one?

File: terraform/testing-suites/xrayreceiver-ec2.tfvars [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+# sample application image to emit the trace data
+data_emitter_image="johnwu20/samplesever:v2.1.0"
+
+# AOC image
+aoc_image_repo="johnwu20/aocimage"
+
+# todo this version needs to be removed, instead version should be received from workflow
+aoc_version="v0.1.1"
+
+region="us-west-2"
```
Comment (wyTrivail) [inline terraform/testing-suites/xrayreceiver-ec2.tfvars:15]: region isn't needed.

File: terraform/testing-suites/xrayreceiver-ecs.tfvars [hunk 6]
```diff
@@ -1,14 +1,16 @@
...
-# todo this is config needs to be removed once we have statsd built in the aoc image
-aoc_image_repo="josephwy/awscollector"
+# AOC image
+aoc_image_repo="johnwu20/aocimage"
 
 # todo this version needs to be removed, instead version should be received from workflow
-aoc_version="xray"
+aoc_version="v0.1.1"
 
+region="us-west-2"
```
Comment (wyTrivail) [inline terraform/testing-suites/xrayreceiver-ecs.tfvars:16]: same as above

**Review by wyTrivail — COMMENTED** [review]
just some small comments, thx

File: terraform/ecs/main.tf (hunk 1 above)
Comment (JohnWu20) [reply terraform/ecs/main.tf:173]: Sure

**Pull request merged**